pkgs/by-name/sk/skyscraper/package.nix

```
@@ -0,0 +1,60 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  qt5,
+  p7zip,
+  python3,
+  installShellFiles,
+
+  # Whether to compile with XDG support
+  # (See: https://gemba.github.io/skyscraper/XDG/)
+  enableXdg ? false,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "skyscraper";
+  version = "3.14.0";
+
+  src = fetchFromGitHub {
+    owner = "Gemba";
+    repo = "skyscraper";
+    rev = "refs/tags/${finalAttrs.version}";
+    hash = "sha256-UgPo9qEsYYYl2RgdPtPmToAruPKuvJ3FSgVrXa94kdA=";
+  };
+
+  strictDeps = true;
+
+  nativeBuildInputs = [
+    qt5.wrapQtAppsHook
+    qt5.qmake
+    installShellFiles
+  ];
+
+  buildInputs = [ python3 ];
+
+  postPatch = lib.optionalString enableXdg ''
+    substituteInPlace skyscraper.pro --replace-fail "#DEFINES+=XDG" "DEFINES+=XDG"
+  '';
+
+  postInstall = ''
+    installShellCompletion --cmd Skyscraper \
+      --bash supplementary/bash-completion/Skyscraper.bash
+  '';
+
+  preFixup = ''
+    qtWrapperArgs+=(--prefix PATH : ${lib.makeBinPath [ p7zip ]})
```

```suggestion
    qtWrapperArgs+=(--prefix PATH : ${lib.makeBinPath [ p7zip ]})
    patchShebangs $out/bin/*.py
```
Seems it didn't do this automatically.

---

I marked Python scripts as executable since it's required for patchShebangs.